### ISSUE TITLE
feature: allow listing files on YAFFS

### ISSUE THREAD

**Issue by robimarko** [kempniu/yafut#17]
Hi,

Is it possible to list files on YAFFS, that would be rather convenient as its expected that the filename is known

**Comment by kempniu** [comment]
There is a YAFFS API for listing directory contents and I even have a horrible local branch that seems to do the job, but... it triggers some sort of a memory leak in YAFFS code that I have not yet looked into (might be an upstream issue or just some mistake on my behalf).  Having a feature like this would be useful for sanity-checking the YAFFS parameters to be used - if filesystem contents can be properly listed and they look "sane", files can probably also at least be read from that file system; if the filesystem looks empty and there are reasons to believe that it shouldn't be, it probably means that Yafut attempts to use incorrect YAFFS parameters.

Summing up, this should eventually work and it would be a useful feature, but it will take time to clean up the messy code that I drafted and to address the outstanding memory leak issue that might need an upstream fix.

**Comment by kempniu** [comment]
Hi @robimarko, I finally had some time to work on this.

Does #32 (branch `enable-listing-yaffs-file-system-contents`) work for you?

**Comment by robimarko** [comment]
Thanks for this, I just have not had time to test it out on a board

**Comment by kempniu** [comment]
No worries, it would not be fair to expect you to respond promptly after I kept you waiting for like a year :-)

Obviously feel free to let me know (in a separate issue) if there is any trouble with the new code!
